backend/src/api/events.js

```diff
@@ -99,6 +100,66 @@ router.put(
   })
 )
 
+// check in to an event
+router.put(
+  '/:eventId/attendees',
+  errorWrap(async (req, res) => {
+    const data = req.body
+    const eventId = req.params.eventId
+    const attendeeEmail = data.email
+
+    const event = await Event.findById(eventId)
+    if (event) {
+      let attendee = await Attendee.findOne({ email: attendeeEmail })
+
+      // create new attendee if not in db
+      if (!attendee) {
+        const newAttendee = new Attendee({
+          name: data.name,
+          email: attendeeEmail,
+          year: data.year
+        })
+        await newAttendee.save()
+        attendee = newAttendee
+      }
+
+      // prevent duplicate check-ins
+      if (event.attendees.includes(attendeeEmail)) {
+        res.json({
+          code: 200,
+          message: 'Already Checked In',
+          success: true
+        })
+      }
+
+      // update event attendees and attendee events
+      event.attendees.push(attendeeEmail)
+      attendee.attendedEvents.push(eventId)
+
+      // handle tardiness
+      if (attendeeIsLate(event)) {
+        event.lateAttendees.push(attendeeEmail)
+        attendee.lateEvents.push(eventId)
+      }
+
+      await event.save()
+      await attendee.save()
+
+      res.json({
+        code: 200,
+        message: 'Successfully Checked In',
+        success: true
+      })
+    } else {
```

Comment:
might be more readable if you did:

```
if (!event) {
    return 404
}
else {
// do whatever
}
```

but it's not really a huge deal

Reply:
ok